---
title: "CpG methylation data analysis results"
author: "Kim Reijntjens"
date: "2022-10-03"
output: pdf_document:
    toc: true 

header-includes:
 \usepackage{float}
 \floatplacement{figure}{H}
---


# Introduction

DNA methylation is a process plays a part in gene regulation. Methylation involves the addition or removal of methyl groups to or from the bases and sugars in DNA. methylation occurs on the cytosine of CG locations in the DNA. methylated CG sequences also called CpG sequences are concentrated in CpG-rich regions called CpG islands, located at the 5'ends of genes. 
Methylation prevents the formation of certain base pairs, and thereby the accessibility for interactions with other components. methylation adds a hydrophobic character to some of the tRNA regions. that may be important for their interaction with proteins that regulate synthesis.
(Klug,William,Cummings,Michael,Spencer,Charlotte,Michael,Palladino)
(Tymoczko, 2015)



## Relavance of the project

Because alterations in the genome and activity of the genes are associated with common diseases such as cancer or asthma, it is important to know if smoking can be the cause of this. If this is the case than in further research there can be looked at which CpG sites are affected by smoking, and if there is a relation with the affected CpG sites and the ones associated with diseases like cancer and asthma. 
(Klug,William,Cummings,Michael,Spencer,Charlotte,Michael,Palladino)

## 1.1 Goal  

Can CpG methylation show a relation with smoking, based on a prediction whether a patient is smoking or non-smoking using differences in CpG site values.


## Data explanation
For this we use the dataset of 683 patients. the dataset has 683 rows and 24 columns containing the patients; age, gender, smoking status and 20 CpG site values. 
The dataset was already compressed to 20 CpG site values sites per sample. The original dataset was composed for a study titled "Differential DNA methylation in Rheumatoid arthritis"
(NCBI series GSE42861) Where the original dataset contained 485577 rows with methylation data per sample 
(NCBI)(KAGGLE)


# Loading the data


```{r, echo = FALSE}

library(pander)
patient_data <- read.csv(file = "data/Smoker_Epigenetic_df.csv")

pander::pander(head(patient_data), caption = " An overview from the fist lines of the data")
```
The first four rows are information about the patient and the remaining 20 are genetic information about the CpG sites of the patients. GSM is an identification number per patient, NCBI shows the full data that was collected for that patient. Smoking status tell us whether a patient is currently smoking or not. Gender and age are self-explanatory. The following 20 column are the CpG site values, these values show a percentage of methylation in these rows. Which can result in a repressed gene activity. We deleted the GMS column because is hold unique row identifiers. Fitting a tree with unique row identifiers in a dataset will split every single row in one node, which will give you a high predictive value. > This will cause overfitting.

```{r, echo = FALSE}
pander::pander(summary(patient_data), caption = "Summary with basic statistics about the data colums")
```

```{r, echo=FALSE, include=FALSE}

#extra changes that are made in the code but will not show up in the results
#these do need to be added in this code chuck otherwise the graphs will be different
library(dplyr)
library(tidyr)
library(ggplot2)
library(gridExtra)
library(grid)
library(tidyverse)

patient_data <- patient_data %>% drop_na()
patient_data <- patient_data %>% mutate(Gender=recode(Gender, 
                         " f"="female",
                         " m"="male"))
patient_data <- patient_data[-1]

```

When we look at the summary of the patient data we see in the column of the methylation data that there are 62 missing values. If we scroll true the data quickly we see that there are some rows who miss all CpG values, those are the ones that show up as 62 missing in every CpG values. these rows will be deleted because it gives us no information for the CpG sites.

# Data exploratory analysis

```{r, echo=FALSE, figures-side, fig.show="hold",fig.cap=" Comparison distribution smoking status and gender", fig.width = 14}



p1 <- ggplot(data=patient_data, aes(Smoking.Status) ) +
  geom_bar(fill=c('pink', "royalblue"),
           color="black", alpha=0.3) +
  ggtitle("number of smoking and non-smoking patients")+
  ylab("number of patients") 

p2 <-  ggplot(data=patient_data, aes(Gender) ) +
  geom_bar(fill=c('pink',"royalblue"),color="black", alpha=0.3) +
  ggtitle("number of male and female patients")+
  ylab("number of patients") 

grid.arrange(p1, p2, ncol = 2)



```


In the figure above we see a distribution overview of the gender and smoking status of the patients. We see a high number of patients are female and the majority is smoking. We will not remove data to get an even distribution of the data, but we do need to keep this in mind when using data mining. 


```{r, echo=FALSE,out.width= "100%", out.height="50%", fig.cap="CpG values in percentages comparison of males and females "}

long_data <- pivot_longer(data = patient_data, cols = 4:23, names_to = "body_part", values_to = "size")

long_data %>%  ggplot(aes(x = size,  colour = Gender)) +
    geom_density(show.legend = TRUE) + 
  ggtitle("CpG values of males and females ") +
   facet_wrap(~body_part, ncol = 7) +    scale_color_manual(values=c("deeppink3", "royalblue") ) 
```



```{r, echo=FALSE,out.width= "100%", out.height="50%", fig.cap="CpG values in percentages comparison of smoking and non smoking patients"}
long_data <- pivot_longer(data = patient_data, cols = 4:23, names_to = "body_part", values_to = "size")

long_data %>%  ggplot(aes(x = size,  colour = Smoking.Status)) +
    geom_density(show.legend = TRUE) +
  ggtitle("CpG values of smoking and non smoking patients ") +
   facet_wrap(~body_part, ncol = 5) +    scale_color_manual(values=c("red", "green")) 
```


As you can see in figure 2, the male and female CpG values are very different from each other. 
the green and red curve are on top of each other which means that there is not much difference in the CpG values for smoking and non smoking patients. This gives us already a lot of information for the research goal. The bimodal double curve in this graph stands for male and female, we observed that in figure 2 where we compared the males and female.


```{r, echo=FALSE,  fig.show="hold",fig.cap="Title:... ", fig.width = 14, fig.cap="underlying groups shwon using PCA"}
df <- subset(patient_data, select = c(4,23) )
row.names(df) <- paste(patient_data$Gender, row.names(df), sep="_") 
df$Gender <- NULL


df_pca <- prcomp(df)

#plot(df_pca$x[,1], df_pca$x[,2])

df_out <- as.data.frame(df_pca$x)
df_out$group <- sapply( strsplit(as.character(row.names(df)), "_"), "[[", 1 )


p<-ggplot(df_out,aes(x=PC1,y=PC2,color=group ))
p<-p+geom_point() +    scale_color_manual(values=c("pink", "royalblue")) +
ggtitle("PCA relations male and female ")
p

#------------------------------------------------------------------------------

df <- subset(patient_data, select = c(4,23) )
row.names(df) <- paste(patient_data$Smoking.Status, row.names(df), sep="_") 
df$Smoking.Status <- NULL

#head(df)

df_pca <- prcomp(df)

#plot(df_pca$x[,1], df_pca$x[,2])

df_out <- as.data.frame(df_pca$x)
df_out$group <- sapply( strsplit(as.character(row.names(df)), "_"), "[[", 1 )
#head(df_out)

p<-ggplot(df_out,aes(x=PC1,y=PC2,color=group ))
p<-p+geom_point() +    scale_color_manual(values=c("red", "green")) +
ggtitle("PCA relations smoking and non-smoking ")
p
```
A principal component analysis gives an extra confirmation of the underlying groups that we saw in the CpG value differences. 
The PCA aims to show pattern in multivariate data. In figure 4 and 5 we see two groups that show a relation in the data. There is a clear difference in males and females. The smokers and non smokers are distributed in both these groups.

# Discussion

Male and female methylation is shown to be different, we cant go deeper in detail in explaining these differences because we do not know the associated genes to the CpG locations in our dataset
 
# Conclusion 

The goal was to understand the given dataset and to clean the data. it is yet to discover if CpG methylation can show a relation with smoking, based on a prediction whether a patient is smoking or non-smoking using differences in CpG site values.
The data shows good prospect for the use of machine learning because of the the patterns and correlation that is found in the data. We already saw that there is a difference in CpG values for gender. Smoking status had not yet shown a clear relation with the CpG values. This will be researched further using machine learning.



# Referances 

- GSE42861. (n.d.). Retrieved from ncbi: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE42861
Klug, W. S., Cummings, M. R., Spencer, C. A., & Palladino, M. A. (n.d.). concepts of genetics.
- thomaskonstantin. (n.d.). Retrieved from kaggle: (https://www.kaggle.com/datasets/thomaskonstantin/cpg-values-of-smoking-and-non-smoking-patients
- Tymoczko, J. L. (2015). Biochemistery: A short course. macmilian publishers.



